src/modules/transaction/components/TxComposer/index.js

```diff
@@ -40,47 +50,61 @@ const TxComposer = ({
     prioritiesLoadError,
     loadingPriorities,
   ] = useTransactionPriority();
-  const [module, command] = splitModuleAndCommand(formProps.moduleCommand);
+
   const transactionJSON = {
     module,
     command,
-    nonce: wallet.sequence?.nonce,
-    fee: 0,
-    senderPublicKey: wallet.summary?.publicKey,
+    nonce: auth?.data?.nonce,
+    fee: formProps.extraCommandFee || 0,
+    senderPublicKey: pubkey,
     params: commandParams,
     signatures: [],
   };
 
-  const status = useTransactionFeeCalculation({
-    token,
-    wallet,
+  const { minimumFee, components, transactionFee } = useTransactionFee({
     selectedPriority,
-    priorityOptions,
     transactionJSON,
+    isFormValid: formProps.isFormValid,
+    senderAddress: address,
+    extraCommandFee: formProps.extraCommandFee,
   });
 
   useEffect(() => {
     if (typeof onComposed === 'function') {
-      onComposed(status, formProps, { ...transactionJSON, fee: toRawLsk(status.fee.value) });
+      onComposed({}, formProps, {
+        ...transactionJSON,
+        fee: toRawLsk(transactionFee),
+      });
     }
   }, [selectedPriority, transactionJSON.params]);
 
-  const minRequiredBalance = getMinRequiredBalance(transactionJSON, status.fee);
+  const minRequiredBalance = getMinRequiredBalance(transactionJSON, transactionFee);
   const { recipientChain, sendingChain } = formProps;
 
-  const composedFees = {
-    Transaction: getFeeStatus({ fee: status.fee, token, customFee }),
-    Initialisation: getFeeStatus({ fee: status.fee, token, customFee }),
-  };
-
-  if (sendingChain && recipientChain && sendingChain.chainID !== recipientChain.chainID) {
-    composedFees.CCM = getFeeStatus({ fee: status.fee, token, customFee });
-  }
+  const composedFees = [
+    {
+      title: 'Transaction',
+      value: getFeeStatus({
+        fee: Number(changeDemnom(transactionFee, formProps.fields.token)),
+        token,
+        customFee,
+      }),
+      components,
+    },
+    {
+      title: 'Message',
+      value: getFeeStatus({
+        fee: Number(fromRawLsk(transactionJSON.params.messageFee || 0)),
+        token,
+        customFee,
+      }),
+      isHidden: !transactionJSON.params.messageFee,
+      components: [],
+    },
```

Comment:
Since composed fee value is done in the generic component `<TxComposer />`, that was the reason I added the extra property `isHidden` which dependent on the fact if a transaction has a messageFee then it would be enabled. Meaning for now only send token for a cross chain transfer would show this value since the value would exist in that scenario.

Reply:
my comment was more towards the code execution. Imagine we are doing validator registration, then the above logic would still be executed which it shouldn't? i.e `MessageFee` is only be executed for cross chain token transfer?

Reply:
Yhea actually that's what is happening the message fee is provided by the send form not from within the TxComposer